openwisp_radius/tests/test_api/test_freeradius_api.py

```diff
@@ -1437,6 +1439,185 @@ def test_authorize_counters_exception_handling(self):
+                "username": user.username,
+                "unique_id": "test_session_1",
+                "stop_time": None,
+            }
+        )
+        self._create_radius_accounting(**acct_data)
+
+        # Authorization should succeed
+        response = self._authorize_user(auth_header=self.auth_header)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["control:Auth-Type"], "Accept")
+
+    @mock.patch.object(app_settings, "ADDITIONAL_RADIUS_CHECKS", ["Simultaneous-Use"])
+    def test_authorize_simultaneous_use_exceeds_limit(self):
+        """Test that user is rejected when exceeding simultaneous use limit"""
+        user = self._get_org_user().user
+        group = user.radiususergroup_set.first().group
+        self._create_radius_groupcheck(
+            group=group,
+            groupname=group.name,
+            attribute="Simultaneous-Use",
+            op=":=",
+            value="1",
+        )
+
+        # Create an open session (at the limit of 1)
+        acct_data = self.acct_post_data
+        acct_data.update(
+            {
+                "username": user.username,
+                "unique_id": "test_session_1",
+                "stop_time": None,  # Open session
+            }
+        )
+        self._create_radius_accounting(**acct_data)
+
+        # Authorization should be rejected
+        response = self._authorize_user(auth_header=self.auth_header)
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response.data["control:Auth-Type"], "Reject")
+        self.assertEqual(
+            response.data["Reply-Message"], "You are already logged in - access denied"
+        )
+
+    @mock.patch.object(app_settings, "ADDITIONAL_RADIUS_CHECKS", ["Simultaneous-Use"])
+    def test_authorize_simultaneous_use_zero_limit_allows_all(self):
+        """Test that zero limit allows unlimited simultaneous sessions"""
+        user = self._get_org_user().user
+        group = user.radiususergroup_set.first().group
+        self._create_radius_groupcheck(
+            group=group,
+            groupname=group.name,
+            attribute="Simultaneous-Use",
+            op=":=",
+            value="0",
+        )
+
+        acct_data = self.acct_post_data
+        acct_data.update(
+            {
+                "username": user.username,
+                "stop_time": None,
+                "session_time": 1,
+                "input_octets": 0,
+                "output_octets": 0,
+            }
+        )
+        # Create multiple open sessions
+        for i in range(3):
+            acct_data["unique_id"] = f"test_session_{i}"
+            self._create_radius_accounting(**acct_data)
+
+        # Authorization should still succeed
+        response = self._authorize_user(auth_header=self.auth_header)
+        self.assertEqual(response.status_code, 200)
+        print(response.data)
+        self.assertEqual(response.data["control:Auth-Type"], "Accept")
```

Comment:
This probably can also be squashed as you simply need to remove the group check.